src/main/java/com/igot/cb/service/PromotionalContentServiceImpl.java

```diff
@@ -0,0 +1,349 @@
+            setFailedResponse(response, errMsg);
+            return response;
+        }
+        try {
+            processAndUpdateUserGroupDetailstoDB(userGroupDetails, response);
+        } catch (Exception e) {
+            log.error("Error while upserting access settings", e);
+            setFailedResponse(response, "Failed to create access settings: " + e.getMessage(),
+                    HttpStatus.INTERNAL_SERVER_ERROR);
+            return response;
+        }
+        return response;
+    }
+
+    /**
+     * Processes user group details, generates UUIDs, and saves to database.
+     * Converts criteria to BitSets via migration service before storage.
+     *
+     * @throws Exception if processing or database operation fails
+     */
+    private void processAndUpdateUserGroupDetailstoDB(Map<String, Object> userGroupDetails, ApiResponse response) throws Exception {
+        Map<String, Object> createPayloadWithUuid = createUserGroupIds(userGroupDetails);
+        Map<String, Object> accessRuleData = new HashMap<>();
+        accessRuleData.put(Constants.CONTEXT_ID, userGroupDetails.get(Constants.CONTENT_ID));
+        accessRuleData.put(Constants.CONTEXT_DATA, objectMapper.writeValueAsString(createPayloadWithUuid));
+        accessRuleData.put(Constants.IS_ARCHIVED, false);
+        if (accessSettingMigrationService.processAccessSettingRule(accessRuleData)) {
+            cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD_COURSE,
+                    Constants.PROMOTIONAL_CONTENT_RULES, accessRuleData);
+            response.getResult().put(Constants.MSG, Constants.PROMOTIONAL_CONTENT_CREATED_RULES);
+            Map<String, Object> payload = new HashMap<>();
+            payload.put(Constants.ACCESS_CONTROL, createPayloadWithUuid.get(Constants.ACCESS_CONTROL));
+            response.getResult().putAll(payload);
+        } else {
+            log.error("Failed to process access setting rule");
+            setFailedResponse(response, "Failed to process access setting rule to id-map",
+                    HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Sets error response with BAD_REQUEST status.
+     */
+    private void setFailedResponse(ApiResponse response, String errorMessage) {
+        setFailedResponse(response, errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    /**
+     * Sets error response with custom HTTP status.
+     */
+    private void setFailedResponse(ApiResponse response, String errorMessage, HttpStatus httpStatus) {
+        response.getParams().setStatus(Constants.FAILED);
+        response.setResponseCode(httpStatus);
+        response.getParams().setErrMsg(errorMessage);
+    }
+
+    /**
+     * Generates UUIDs for user groups without IDs.
+     * Modifies payload in-place and returns it.
+     */
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> createUserGroupIds(Map<String, Object> payload) {
```

Comment:
criteria list needs to be added into the user group and criteria key value pair needs to be added into the criteria list.. 
As of now, incoming payload will contain only the designation